### PR TITLE
[TAS-1216] ✨ Define book cover for PDF file

### PR DESCRIPTION
**Pull request by AuroraHuang22**
When a user uploads a PDF and a JPG/PNG image simultaneously, the first uploaded image should be set as the default cover image and written into the ISCN's thumbnail field.

QA flow:

1. Navigate to /new to register an ISCN.
2. Upload a PDF and an image.
3. Complete the registration process.
4. Verify in the downloaded JSON file if the thumbnail field has been populated.

### CODE REVIEW

File: components/IscnUploadForm.vue [hunk 1]
```diff
@@ -837,6 +913,11 @@ export default class IscnUploadForm extends Vue {
 
     try {
       this.uploadStatus = 'uploading';
+
+      if (this.checkUploadFileTypeIsPDF()) {
```
Comment (williamchong) [inline components/IscnUploadForm.vue:917]: It seems the function does not simply check `is PDF`, but `PDF and not EPUB`, may be split 2 logic?
Sth like `this.checkUploadFileTypeContainsPDF() && !this.checkUploadFileTypeContainsEPUB()` (actually simple `.find` would do, we might not even need a function`

Comment (AuroraHuang22) [reply components/IscnUploadForm.vue]: @AuroraHuang22
[🎨 Apply simplification suggestion](https://github.com/likecoin/app-like-co/pull/457/commits/79cd883bbdabd207ea9772326f58a5c90d271592)

File: components/IscnUploadForm.vue [hunk 2]
```diff
@@ -813,6 +820,75 @@ export default class IscnUploadForm extends Vue {
...
+      if (IMAGE_MIME_TYPES.includes(file.fileType)) {
+        const existingData =
+          this.sentArweaveTransactionInfo.get(file.ipfsHash) || {}
+        if (existingData.arweaveId) {
+          this.addToEpubMetadataList(file.ipfsHash, existingData.arweaveId)
+          return
+        }
+        const transactionHash =
+        // eslint-disable-next-line no-await-in-loop
+          existingData.transactionHash || (await this.sendArweaveFeeTx(file))
```
[the remark this answers is not in the document]
Comment (AuroraHuang22) [reply components/IscnUploadForm.vue]: [🎨 Apply simplification suggestion](https://github.com/likecoin/app-like-co/pull/457/commits/79cd883bbdabd207ea9772326f58a5c90d271592)

File: components/IscnUploadForm.vue [hunk 3]
```diff
@@ -813,6 +820,75 @@ export default class IscnUploadForm extends Vue {
...
+    })
+  }
+
+  async setEbookCoverFromImages() {
+    if (
+      this.epubMetadataList[0] &&
+      this.epubMetadataList[0].thumbnailArweaveId
+    ) {
+      return
+    }
```
[the remark this answers is not in the document]
Comment (AuroraHuang22) [reply components/IscnUploadForm.vue]: [🎨 Apply simplification suggestion](https://github.com/likecoin/app-like-co/pull/457/commits/79cd883bbdabd207ea9772326f58a5c90d271592)

File: components/IscnUploadForm.vue [hunk 4]
```diff
@@ -813,6 +820,75 @@ export default class IscnUploadForm extends Vue {
...
+    return hasPDF;
+  }
+
+
+  addToEpubMetadataList(ipfsHash: string, arweaveId: string) {
+    this.epubMetadataList.push({
+      thumbnailIpfsHash: ipfsHash,
+      thumbnailArweaveId: arweaveId,
+    })
+  }
```
[the remark this answers is not in the document]
Comment (AuroraHuang22) [reply components/IscnUploadForm.vue]: [🎨 Apply simplification suggestion](https://github.com/likecoin/app-like-co/pull/457/commits/79cd883bbdabd207ea9772326f58a5c90d271592)

File: components/IscnUploadForm.vue [hunk 5]
```diff
@@ -813,6 +820,75 @@ export default class IscnUploadForm extends Vue {
...
+    let hasPDF = false;
+    // eslint-disable-next-line no-restricted-syntax
+    for (const file of this.fileRecords) {
+      if (file.fileType === 'application/epub+zip') {
+        return false;
+      }
+      if (file.fileType === 'application/pdf') {
+        hasPDF = true;
+      }
+    }
```
[the remark this answers is not in the document]
Comment (AuroraHuang22) [reply components/IscnUploadForm.vue]: [🎨 Apply simplification suggestion](https://github.com/likecoin/app-like-co/pull/457/commits/79cd883bbdabd207ea9772326f58a5c90d271592)

File: components/IscnUploadForm.vue [hunk 6]
```diff
@@ -813,6 +820,75 @@ export default class IscnUploadForm extends Vue {
...
+
+  async setEbookCoverFromImages() {
+    if (
+      this.epubMetadataList[0] &&
+      this.epubMetadataList[0].thumbnailArweaveId
+    ) {
+      return
+    }
+    // eslint-disable-next-line no-restricted-syntax
+    for (const file of this.fileRecords) {
```
[the remark this answers is not in the document]
Comment (AuroraHuang22) [reply components/IscnUploadForm.vue]: [🎨 Use promise.all to get arweaveId](https://github.com/likecoin/app-like-co/pull/457/commits/9cfa8a3306d3caeb1e797fcd14042a65b704c0de)

[the remark this answers is not in the document]
Comment (AuroraHuang22) [reply components/IscnUploadForm.vue]: [🎨 Use traditional for loop to get arweaveId](https://github.com/likecoin/app-like-co/pull/457/commits/bc8ddb71d0c95fc978de84c76fa735d04663cd5d)

[the remark this answers is not in the document]
Comment (AuroraHuang22) [reply components/IscnUploadForm.vue]: [🎨 Switch from return to break in setEbookCoverFromImages](https://github.com/likecoin/app-like-co/pull/457/commits/83190c931e846d9485fa6a28980839422c5b4961)

**Review by williamchong — APPROVED** [review]

**Pull request merged**